e2etests/pkg/config/from_containers.go

```diff
@@ -25,33 +32,57 @@ type PeersConfig struct {
 	Secrets []corev1.Secret
 }
 
+func (pc PeersConfig) Peers() []Peer {
+	return append(pc.PeersV4, pc.PeersV6...)
+}
+
+func (pc PeersConfig) FRRK8sNeibghbor() []frrk8sv1beta1.Neighbor {
+	res := make([]frrk8sv1beta1.Neighbor, 0)
+	for _, p := range pc.PeersV4 {
+		res = append(res, p.Neigh)
+	}
+	for _, p := range pc.PeersV6 {
+		res = append(res, p.Neigh)
+	}
+	return res
+}
+
 // PeersForContainers returns two lists of Peers, one for v4 addresses and one for v6 addresses.
 func PeersForContainers(frrs []*frrcontainer.FRR, ipFam ipfamily.Family) PeersConfig {
+	return PeersForContainersWithOptions(frrs, ipFam, AllowAllDisabled, GracefulRestartDisabled)
+}
+
+func PeersForContainersWithOptions(frrs []*frrcontainer.FRR, ipFam ipfamily.Family, allowAll, gracefulRestart bool) PeersConfig {
```

Comment:
I'd use the functional options pattern here  https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
you can have one function that takes the generated PeersConfig and iterates applying the changes. Or you can have a neighborModifier. I don't like the `WithOptions` wrapper that is `WithOptionsThatINeedOnlyForThisTest`.

Reply:
done